### PR TITLE
Handle override upgrades

### PR DESCRIPTION
**Pull request by shomron**
Signed-off-by: Oren Shomron <shomron@gmail.com>

### CODE REVIEW

File: pkg/app/override.go [hunk 1]
```diff
@@ -28,10 +29,13 @@ import (
 const (
 	// overrideKind is the override resource type.
 	overrideKind = "ksonnet.io/app-override"
-	// overrideVersion is the version of the override resource.
+	// OverrideVersion is the version of the override resource.
```
Comment (bryanl) [inline pkg/app/override.go]: The comment changed but the const did not. Was this intentional?

File: pkg/app/override.go [hunk 2]
```diff
@@ -28,10 +29,13 @@ import (
...
 	overrideVersion = "0.3.0"
 )
 
+// OverrideSemVersion is a parsed semver.Version representing the current Override version.
+var overrideSemVersion = semver.MustParse(overrideVersion)
```
Comment (bryanl) [inline pkg/app/override.go]: The godoc doesn't match the variable.

File: pkg/app/override.go (hunk 1 above)
Comment (shomron) [reply pkg/app/override.go]: No, there was a back and forth and I missed the comment when I changed it back. Thanks!

File: pkg/app/override.go (hunk 2 above)
Comment (shomron) [reply pkg/app/override.go]: ...same issue as above.

**Pull request merged**